core/src/services/gridfs/backend.rs

```diff
@@ -0,0 +1,299 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use async_trait::async_trait;
+use futures::{AsyncWriteExt, StreamExt};
+use mongodb::bson::doc;
+use mongodb::options::{ClientOptions, GridFsBucketOptions, GridFsFindOptions};
+use mongodb::GridFsBucket;
+use std::fmt::{Debug, Formatter};
+use tokio::sync::OnceCell;
+
+use crate::raw::adapters::kv;
+use crate::*;
+
+#[doc = include_str!("docs.md")]
+#[derive(Default)]
+pub struct GridFsBuilder {
+    connection_string: Option<String>,
+    database: Option<String>,
+    bucket: Option<String>,
+    chunk_size: Option<u32>,
+    root: Option<String>,
+}
+
+impl Debug for GridFsBuilder {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("GridFsBuilder")
+            .field("connection_string", &self.connection_string)
```

Comment:
Please avoid exposing the connection string in Debug as it may contain sensitive information.